From b8cdf84039078f6162f76d4f5fe722dacce9ef7d Mon Sep 17 00:00:00 2001
From: Adam Green <greena88@gmail.com>
Date: Mon, 23 Apr 2018 04:20:09 +0100
Subject: [PATCH] Remove settings toggle for 'sync playback to display' as this
 is handled by amcodec and enabling this setting can cause bad video playback,
 also rename xml settings value so this setting will default to off for users
 who are upgrading and lose the ability to toggle the option off.

---
 system/settings/settings.xml | 5 -----
 xbmc/settings/Settings.cpp   | 2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/system/settings/settings.xml b/system/settings/settings.xml
index 6b58b5f..baf0851 100755
--- a/system/settings/settings.xml
+++ b/system/settings/settings.xml
@@ -46,11 +46,6 @@
           </constraints>
           <control type="list" format="string" />
         </setting>
-        <setting id="videoplayer.usedisplayasclock" type="boolean" label="13510" help="36166">
-          <level>1</level>
-          <default>false</default>
-          <control type="toggle" />
-        </setting>
         <setting id="videoplayer.errorinaspect" type="integer" label="22021" help="36170">
           <level>2</level>
           <default>0</default>
diff --git a/xbmc/settings/Settings.cpp b/xbmc/settings/Settings.cpp
index 60aff91..ebe915d 100644
--- a/xbmc/settings/Settings.cpp
+++ b/xbmc/settings/Settings.cpp
@@ -143,7 +143,7 @@ const std::string CSettings::SETTING_VIDEOPLAYER_AUTOPLAYNEXTITEM = "videoplayer
 const std::string CSettings::SETTING_VIDEOPLAYER_SEEKSTEPS = "videoplayer.seeksteps";
 const std::string CSettings::SETTING_VIDEOPLAYER_SEEKDELAY = "videoplayer.seekdelay";
 const std::string CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE = "videoplayer.adjustrefreshrate";
-const std::string CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK = "videoplayer.usedisplayasclock";
+const std::string CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK = "videoplayer.usedisplayasclock.null";
 const std::string CSettings::SETTING_VIDEOPLAYER_ERRORINASPECT = "videoplayer.errorinaspect";
 const std::string CSettings::SETTING_VIDEOPLAYER_STRETCH43 = "videoplayer.stretch43";
 const std::string CSettings::SETTING_VIDEOPLAYER_TELETEXTENABLED = "videoplayer.teletextenabled";
-- 
2.7.4

